src/seedu/addressbook/data/person/Assessment.java

```diff
@@ -12,26 +12,38 @@
 public class Assessment {
 
     public static final String EXAM_NAME_EXAMPLE = "CG2271 Midterms";
+    public static final String MESSAGE_ASSESSMENT_CONSTRAINTS = "Assessment name can contain $";
+    public static final String ASSESSMENT_VALIDATION_REGEX = "^((?!\\$).)*$";
 
     private String examName;
-    private List<Grades> grade;
+    private Map<Person, Grades> grade;
 
```

Comment:
Hmm.  Since you don't need to the person to be editable when assigning his grades right?
I'm not sure if you should be mapping Person -> Grades or is it supposed to be ReadOnlyPerson -> Grades?